velox/dwio/parquet/writer/Writer.cpp

```diff
@@ -223,7 +227,21 @@ dwio::common::StripeProgress getStripeProgress(
  *
  * This method assumes each input `ColumnarBatch` have same schema.
  */
-void Writer::write(const VectorPtr& data) {
+void Writer::write(const VectorPtr& dataToWrite) {
+  VectorPtr data = dataToWrite;
+  if (schema_) {
```

Comment:
Add a check that schema_ and dataToWrite->type() are equivalent.

Maybe add a test for nested structs. Make sure name for nested struct fields are used properly.